### DISCUSSION TITLE
Generic ESP32 2.4G PWM target

### DISCUSSION THREAD

**Discussion by JyeSmith** [ExpressLRS/ExpressLRS#1929] [Ideas]
At the moment we have Frank but not a dedicated esp32 pwm target.  Im not a wing/pwm guy but thought I would get the ball rolling.  There have already been a couple of diy suggestions and we also have manufacturers wishing to do a 2.4 True Diversity PWM Rx.

Below is a 14 ch target I have just confirmed on the bench is working.  It follows the True Diversity target with the exception of moving MISO to give us another channel.

I personally would like to retain gpio0 as a button.  A dedicated pwm rx could also have usb flashing built in, and the boot button might be useful.

Sensors... I have NFI.  Can we keep 12/13 channels, and then mod 2 later on the pin header later?

Lets hear some ideas and get a target finalised 😃 

```
{
    "radio_miso": 35,
    "radio_mosi": 32,
    "radio_rst": 26,
    "radio_sck": 25,

    "radio_busy": 36,
    "radio_dio1": 37,
    "radio_nss": 27,

    "radio_busy_2": 39,
    "radio_dio1_2": 34,
    "radio_nss_2": 13,

    "power_txen": 14,
    "power_txen_2": 15,
    "power_lna_gain": 12,

    "power_min": 0,
    "power_high": 3,
    "power_max": 3,
    "power_default": 3,
    "power_control": 0,
    "power_values": [-10,-6,-3,1],

    "led_rgb": 22,
    "led_rgb_isgrb": true,
    "ledidx_rgb_status": [0],
    "ledidx_rgb_boot": [0],

    "vbat": 38,
    "vbat_offset": 12,
    "vbat_scale": 410,
    
    "pwm_outputs": [1, 3, 21, 19,   5, 2, 18, 12,   33, 9, 10,  4,   0, 23]
}

```

**Comment by secretman28** [comment]
nice idea many plane guy wait that to make the move i personaly use matek 10 ch but need crsf to sbus converter too to get the gyro working can be nice to have an option to set the output to sbus in some of the rx target the converter add latency

**Reply by CREASE-gum-EAR** [reply]
Yes add SBUS as a selectable output

**Comment by psguardian** [comment]
I fly both fixed wing & quads (with a heli in storage).
There are people at my flying club eyeing elrs, but need more pwm: large bipes can use 7 servos + throttle. Full house sailplanes 10 servos + throttle. Nitro planes add anther for idle adjust.
Many of these pilots despise the idea of computer aided safeties like as3x & recovery switches / horizon modes like you would see on a larger flight controller with 12-16 pwm. So if they could get a stand alone diversity rx with full compliment of pwm outputs they would move their fleets.

**Reply by CREASE-gum-EAR** [reply]
One option is to put a separate receiver in each extremity of the airplane and map difference channels to each (see the video link here).
https://www.youtube.com/watch?v=oUcMMjpN4u0
But PNP planes come with all servos with leads extending to the center. So the biggest concern is all the amp draw of servos on those big setups. A PDB and other backups are needed but many systems rely on SBUS for the signals and use third party PDBs. There is also countless other solutions. And due to how expensive the planes are, there is piece of mind in backup systems. Even if its a little unfounded with LoRa and flying LOS. But still, its just piece of mind they are looking for. If there was just SBUS out then allot of this could be solved for fixed wing pilots who already have tried and true servo systems.

**Reply by psguardian** [reply]
The issue with splitting channels is that if you do drop reception of one antenna,  you drop half of your controls. Which is a good as losing it all.  Both antennas need to carry the full channel stack.

**Comment by DocTrucker** [comment]
Just in the process of putting together an ep1 dual with CRSF to PWM adaptor for my buggy, so watching this with interest.

The RC car lot seem very brand loyal. There are a few proprietary protocals from Futaba and Sanwa that are popular (eg sbus, ssr) but I would be interested to learn if there are issues that would leave the open CANBUS protocols as the bottle neck in the Rx stick twitch to Motor/Servo reaponse chain. There are more CAN servos coming on the market now and given that a lot of devices like GPS are going to CAN too, I'd expect that to grow.
